src/ResourceManager/KeyVault/Commands.KeyVault/Commands/ManagedStorageAccounts/SetAzureKeyVaultManagedStorageSasDefinition.cs

```diff
@@ -0,0 +1,361 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using Microsoft.Azure.Commands.KeyVault.Models;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Management.Automation;
+using System.Text;
+using System.Xml;
+
+namespace Microsoft.Azure.Commands.KeyVault
+{
+    [Cmdlet( VerbsCommon.Set, CmdletNoun.AzureKeyVaultManagedStorageSasDefinition,
+        SupportsShouldProcess = true,
+        HelpUri = Constants.KeyVaultHelpUri,
+        DefaultParameterSetName = ParameterSetRawSas )]
+    [OutputType( typeof( ManagedStorageSasDefinition ) )]
+    public partial class SetAzureKeyVaultManagedStorageSasDefinition : KeyVaultCmdletBase
+    {
+        private const string ParameterSetRawSas = "RawSas";
+
+        [Parameter( Mandatory = true,
+            Position = 0,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Vault name. Cmdlet constructs the FQDN of a vault based on the name and currently selected environment." )]
+        [ValidateNotNullOrEmpty]
+        public string VaultName { get; set; }
+
+        [Parameter( Mandatory = true,
+            Position = 1,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Key Vault managed storage account name. Cmdlet constructs the FQDN of a managed storage account name from vault name, currently " +
+                          "selected environment and manged storage account name." )]
+        [ValidateNotNullOrEmpty]
+        [Alias( Constants.StorageAccountName )]
+        public string AccountName { get; set; }
+
+        [Parameter( Mandatory = true,
+            Position = 2,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Storage sas definition name. Cmdlet constructs the FQDN of a storage sas definition from vault name, currently " +
+                          "selected environment, storage account name and sas definition name." )]
+        [ValidateNotNullOrEmpty]
+        [Alias( Constants.SasDefinitionName )]
+        public string Name { get; set; }
+
+        [Parameter( Mandatory = true,
+            Position = 3,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Sas definition parameters that will be used to create the sas token.",
+            ParameterSetName = ParameterSetRawSas )]
+        [ValidateNotNull]
+        public Hashtable Parameter { get; set; }
+
+        [Parameter( Mandatory = false,
+            HelpMessage = "Disables the use of sas definition for generation of sas token." )]
+        public SwitchParameter Disable { get; set; }
```

Comment:
This parameter is named along the lines of similar parameter in Set-AzureKeyVaultSecret and Set-AzureKeyVaultKey and carries similar meaning of disabling use of the key vault object. 

---
In reply to: [119732397](https://github.com/Azure/azure-powershell/pull/4061#discussion_r119732397) [](ancestors = 119732397)